### PR TITLE
manage images through webpack

### PR DESCRIPTION
**Pull request by LeoMcA**
- enable loading images from webpack in django
- load all images through webpack
- optimise png/svg in webpack

https://github.com/mozilla/sumo-project/issues/875
https://github.com/mozilla/sumo-project/issues/880

### CODE REVIEW

File: kitsune/sumo/static/sumo/js/main.js [hunk 1]
```diff
@@ -3,7 +3,6 @@ window.k = window.k || {};
 
 (function () {
   k.LAZY_DELAY = 500;  // delay to lazy loading scripts, in ms
-  k.STATIC_URL = $('body').data('static-url');
```
Comment (akatsoulas) [inline kitsune/sumo/static/sumo/js/main.js:6]: There are a couple of references in the `styleguide` too

File: kitsune/sumo/utils.py [hunk 2]
```diff
@@ -350,3 +351,37 @@ def check_for_spam_content(data):
...
+    **{
+        f"{app}/": f"kitsune/{app}/static/{app}/"
+        for app in [
+            app.removeprefix("kitsune.")
+            for app in settings.INSTALLED_APPS
+            if app.startswith("kitsune.")
+        ]
+    },
+}
+_webpack_static_cache = {}
```
Comment (akatsoulas) [inline kitsune/sumo/utils.py:367]: Let's remove caching for now from this PR. Let's have an issue to further investigate this.

Comment (akatsoulas) [reply kitsune/sumo/utils.py:367]: Along with this let's remove the setting too since it won't be used.

File: kitsune/sumo/utils.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 
 from django.conf import settings
 from django.contrib.sites.models import Site
+from django.contrib.staticfiles.templatetags.staticfiles import static as django_static
```
Comment (akatsoulas) [inline kitsune/sumo/utils.py:9]: `from django.templatetags.static import static`

File: kitsune/sumo/utils.py [hunk 4]
```diff
@@ -350,3 +351,37 @@ def check_for_spam_content(data):
...
+    Accepts a shortened path to the source, like:
+    "sumo/img/mozilla-support.svg" rather than "kitsune/sumo/static/sumo/img/mozilla-support.svg"
+    """
+
+    if settings.WEBPACK_USE_STATIC_CACHE and source_path in _webpack_static_cache:
+        return _webpack_static_cache[source_path]
+
+    with open("./dist/source-to-asset.json") as f:
+        source_to_asset = json.load(f)
+        full_path = re.sub(
```
Comment (akatsoulas) [inline kitsune/sumo/utils.py:382]: This feels redundant. We already have the asset under the `django_static(source_path)`. I will continue looking into this though

File: kitsune/sumo/templatetags/jinja_helpers.py [hunk 5]
```diff
@@ -453,6 +453,11 @@ def static(path):
         return ""
 
 
+@library.global_function
+def webpack_static(source_path):
```
[the remark this answers is not in the document]
Comment (LeoMcA) [reply kitsune/sumo/templatetags/jinja_helpers.py:457]: It results in some nasty circular imports in the `models.py` files which need to use `webpack_static`, if they have to import from jinja_helpers. I can move the logic here and do a bunch of inline imports though, up to you

File: kitsune/sumo/utils.py (hunk 2 above)
Comment (LeoMcA) [reply kitsune/sumo/utils.py]: https://github.com/mozilla/sumo-project/issues/874

File: kitsune/sumo/static/sumo/js/main.js (hunk 1 above)
Comment (LeoMcA) [reply kitsune/sumo/static/sumo/js/main.js:6]: The styleguide has a bunch of peculiarities and I'm tackling them in a separate issue: https://github.com/mozilla/sumo-project/issues/894

File: kitsune/sumo/utils.py [hunk 6]
```diff
@@ -350,3 +351,37 @@ def check_for_spam_content(data):
     has_links = has_blocked_link(data)
 
     return is_toll_free or is_nanp_number or has_links
+
+
+WEBPACK_SHORT_PATH_MAPPING = {
```
[the remark this answers is not in the document]
Comment (Sheabae86) [reply kitsune/sumo/utils.py]: > move this at the top of the file

Why

**Review by akatsoulas — APPROVED** [review]
r+wc

**Pull request merged**